src/deepsparse/loggers/async_logger.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Logger wrapper to run log calls asynchronously to not block the main process
+"""
+
+import logging
+from concurrent.futures import Executor, ThreadPoolExecutor
+from typing import Any
+
+from deepsparse.loggers import BaseLogger, MetricCategories
+
+
+__all__ = ["AsyncLogger"]
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AsyncLogger(BaseLogger):
+    """
+    Logger wrapper that forwards log calls to run asynchronously, freeing
+    the main process. No call back/returned future currently provided
+
+    :param logger: logger object to wrap
+    :param max_workers: maximum logging tasks to run in the job pool at once.
+        defaults to 1
+    """
+
+    def __init__(self, logger: BaseLogger, max_workers: int = 1):
+        self.logger = logger
+        self._job_pool: Executor = ThreadPoolExecutor(max_workers=max_workers)
```

Comment:
Option to pass this in to this object? I think server could pass it the thread pool executor it owns to this so they share one. Or do we explicitly want a separate one I guess? How do 2 thread pool executors interact?

Reply:
yes I like this, let's get it added in later. 2 thread pools should be ok depending on usage, can lead to deadlock in certain scenarios. let's come back to this in testing